syncano_cli/sync/commands.py

```diff
@@ -1,92 +1,109 @@
-        context.project.write(context.file)
-
-
-class Watch(six.with_metaclass(CommandContainer)):
-    namespace = SYNC_NAMESPACE
-
-    @classmethod
-    @argument('instance', help="Source instance name", action=EnvDefault,
-              envvar='SYNCANO_INSTANCE')
-    def run(cls, context):
-        """
-        Push configuration to syncano. After that  watch for changes in
-        syncano.yml file and scripts and push changed items to syncano.
-        """
-        context.classes = None
-        context.scripts = None
-        context.all = True
-        context.project.timestamp = 0
-        Push.run(context)
-        LOG.info("Watching for file changes")
-        observer = Observer()
-        observer.schedule(ProjectEventHandler(context), path='.', recursive=True)
-        observer.start()
-        try:
-            while True:
-                time.sleep(1)
-        except KeyboardInterrupt:
-            observer.stop()
-        observer.join()
+
+@click.group()
+@click.pass_context
+def top_sync(context):
+    pass
+
+
+@top_sync.group()
+@click.pass_context
+@click.option('-f', '--file', default='syncano.yml', help=u'Instance configuration file.')
+@click.option('--config', help=u'Account configuration file.')
+@click.option('--key', '--key', default=os.environ.get('SYNCANO_API_KEY', None),
+              help=u'override ACCOUNT_KEY used for authentication.')
+def sync(context, file, config, key):
+    """
+    Command for syncing data - classes and scripts
+    :param context:
+    :param file: file which will be used for syncing
+    :param config: the config path - the cli config will be stored there
+    :param key: the Syncano API key
+    :return:
+    """
+    from syncano_cli.main import ACCOUNT_CONFIG, ACCOUNT_CONFIG_PATH
+    config = config or ACCOUNT_CONFIG_PATH
+    ACCOUNT_CONFIG.read(config)
+    context.obj['file'] = file
+    context.obj['config'] = config
+    context.obj['key'] = key or ACCOUNT_CONFIG.get('DEFAULT', 'key')
+    context.obj['project'] = Project.from_config(context.obj['file'])
+
+
+@sync.command()
+@click.pass_context
+@click.option('-s', '--script', help=u"Pull only this script from syncano", multiple=True)
+@click.option('-c', '--klass', help=u"Pull only this class from syncano", multiple=True)
+@click.option('-a', '--all/--no-all', default=False, help=u"Force push all configuration")
+@click.argument('instance', envvar='SYNCANO_INSTANCE')
+def push(context, scripts, klass, all, instance):
+    """
+    Push configuration changes to syncano.
+    """
+    do_push(context, scripts=scripts, classes=klass, all=all, instance=instance)
+
+
+@sync.command()
+@click.pass_context
+@click.option('-s', '--script', help=u"Pull only this script from syncano", multiple=True)
+@click.option('-c', '--klass', help=u"Pull only this class from syncano", multiple=True)
+@click.option('-a', '--all/--no-all', default=False, help=u"Force push all configuration")
```

Comment:
`--no-all` may not make much sense though (I guess?)
maybe make it a flag? `@click.option('--all', is_flag=True)`